internal/handshake/crypto_setup.go

```diff
@@ -506,11 +507,12 @@ func (h *cryptoSetup) ReadHandshakeMessage() ([]byte, error) {
 	} else {
```

Comment:
On line 488, is there a possibility the `go func() { <-h.isReadingHandshakeMessage` gets stuck somehow?

Reply:
Good catch. That one should probably be protected by a `select` statement as well.